ingresspipe/utils/general.py

```diff
@@ -1,117 +1,41 @@
-        # execute all requests
-        response = service.spreadsheets().batchUpdate(spreadsheetId=kwargs["spreadsheet_id"], body = requests_body).execute()
-        
-        return response
 
+def str2list(_str):
+    if type(_str) == str:
+        return [_str]
+    elif type(_str) == list:
+        return _str
 
-# utilities for schema explorer methods
 
-def expand_curie_to_uri(curie, context_info):
-    """Expand curie to uri based on the context given
-
-    parmas
-    ======
-    curie: curie to be expanded (e.g. bts:BiologicalEntity)
-    context_info: jsonld context specifying prefix-uri relation (e.g. {"bts":
-    "http://schema.biothings.io/"})
-    """
-    # as suggested in SchemaOrg standard file, these prefixes don't expand
-    PREFIXES_NOT_EXPAND = ["rdf", "rdfs", "xsd"]
-    # determine if a value is curie
-    if len(curie.split(':')) == 2:
-        prefix, value = curie.split(":")
-        if prefix in context_info and prefix not in PREFIXES_NOT_EXPAND:
-            return context_info[prefix] + value
-    # if the input is not curie, return the input unmodified
-        else:
-            return curie
+def unlist(_list):
+    if len(_list) == 1:
+        return _list[0]
     else:
-        return curie
+        return _list
 
 
-def expand_curies_in_schema(schema):
-    """Expand all curies in a SchemaOrg JSON-LD file into URI
-    """
-    context = schema["@context"]
-    graph = schema["@graph"]
-    new_schema = {"@context": context,
-                  "@graph": [],
-                  "@id": schema["@id"]}
-    for record in graph:
-        new_record = {}
-        for k, v in record.items():
-            if type(v) == str:
-                new_record[expand_curie_to_uri(k, context)] =  expand_curie_to_uri(v, context)
-            elif type(v) == list:
-                if type(v[0]) == dict:
-                    new_record[expand_curie_to_uri(k, context)] = []
-                    for _item in v:
-                        new_record[expand_curie_to_uri(k, context)].append({"@id": expand_curie_to_uri(_item["@id"], context)})
-                else:
-                    new_record[expand_curie_to_uri(k, context)] = [expand_curie_to_uri(_item, context) for _item in v]
-            elif type(v) == dict and "@id" in v:
-                new_record[expand_curie_to_uri(k, context)] = {"@id": expand_curie_to_uri(v["@id"], context)}
-            elif v == None:
-                new_record[expand_curie_to_uri(k, context)] = None
-        new_schema["@graph"].append(new_record)
-    return new_schema
-
-
-def uri2label(uri, schema):
-    """Given a URI, return the label
-    """
-    return [record["rdfs:label"] for record in schema["@graph"] if record['@id'] == uri][0]
-
-
-def find_duplicates(_list):
-    """Find duplicate items in a list
-    """
-    return set([x for x in _list if _list.count(x) > 1])
\ No newline at end of file
+def visualize(edges, size=None):
```

Comment:
I see we have df_utils module; do you think it's justified to also have viz_utils that contain functions related to visualization - we might add more options in the future there?

Reply:
Yes, we could add a `viz_utils` module, right now, it's just sitting in `general`, and it's probably a good idea to move it out.